tests/lib/test_docker.py

```diff
@@ -0,0 +1,151 @@
+import nestor_api.lib.docker as docker
+import nestor_api.lib.git as git
+import nestor_api.lib.io as io
+from nestor_api.utils.logger import Logger
+
+
+def test_build_already_built(mocker):
+    mocker.patch.object(docker, "has_docker_image", return_value=True)
+    mocker.patch.object(git, "get_last_tag", return_value="1.0.0-sha-a2b3c4")
+    context = {"commitHash": "a2b3c4", "repository": "/path_to/a_git_repository"}
+
+    image_tag = docker.build("my-app", context)
+
+    git.get_last_tag.assert_called_once_with("/path_to/a_git_repository")
+    docker.has_docker_image.assert_called_once_with("my-app", "1.0.0-sha-a2b3c4")
+
+    assert image_tag == "1.0.0-sha-a2b3c4"
+
+
+def test_build(mocker):
+    app_config = {"build": {"variables": {"var1": "val1", "var2": "val2"}}}
+
+    mocker.patch.object(docker, "has_docker_image", return_value=False)
+    mocker.patch.object(git, "get_last_tag", return_value="1.0.0-sha-a2b3c4")
+    mocker.patch.object(config, "get_app_config", return_value=app_config)
+    mocker.patch.object(io, "execute", return_value="")
+
+    context = {"commitHash": "a2b3c4", "repository": "/path_to/a_git_repository"}
+
+    image_tag = docker.build("my-app", context)
+
+    git.get_last_tag.assert_called_once_with("/path_to/a_git_repository")
+    docker.has_docker_image.assert_called_once_with("my-app", "1.0.0-sha-a2b3c4")
+    config.get_app_config.assert_called_once_with("my-app")
+    io.execute.assert_called_once_with(
+        "docker build"
+        " --tag my-app:1.0.0-sha-a2b3c4"
+        " --build-arg var1=val1"
+        " --build-arg var2=val2"
+        " --build-arg COMMIT_HASH=None"
+        " /path_to/a_git_repository"
+    )
+
+    assert image_tag == "1.0.0-sha-a2b3c4"
+
+
+def test_build_failure(mocker):
+    app_config = {"build": {"variables": {"var1": "val1", "var2": "val2"}}}
+
+    mocker.patch.object(docker, "has_docker_image", return_value=False)
+    mocker.patch.object(git, "get_last_tag", return_value="1.0.0-sha-a2b3c4")
+    mocker.patch.object(config, "get_app_config", return_value=app_config)
+
+    exception = subprocess.CalledProcessError(1, "docker build")
+    mocker.patch.object(io, "execute", side_effect=[exception])
+    mocker.patch.object(Logger, "error", return_value="")
+
+    context = {"commitHash": "a2b3c4", "repository": "/path_to/a_git_repository"}
+
+    with pytest.raises(Exception):
+        docker.build("my-app", context)
+
+    git.get_last_tag.assert_called_once_with("/path_to/a_git_repository")
+    docker.has_docker_image.assert_called_once_with("my-app", "1.0.0-sha-a2b3c4")
+    config.get_app_config.assert_called_once_with("my-app")
+    io.execute.assert_called_once_with(
+        "docker build"
+        " --tag my-app:1.0.0-sha-a2b3c4"
+        " --build-arg var1=val1"
+        " --build-arg var2=val2"
+        " --build-arg COMMIT_HASH=None"
+        " /path_to/a_git_repository"
+
+    )
+    Logger.error.assert_called_once_with(
```

Comment:
I'm wondering if this is possible to assert the error raised from `docker.build()` instead of testing that `logger.error()` is called ? What do you think ?

Reply:
It is what is done here : https://github.com/ChauffeurPrive/nestor-api/blob/docker-lib/tests/lib/test_docker.py#L68